RealmSwift/Tests/MapTests.swift

```diff
@@ -0,0 +1,858 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import XCTest
+import RealmSwift
+
+fileprivate extension Map {
+    func addTestObjects(from dictionary: [Key: Value]) where Key: Hashable {
+        dictionary.forEach { (k, v) in
+            self[k] = v
+        }
+    }
+}
+
+class MapTests<M: RealmKeyedCollection, EM: RealmKeyedCollection>: TestCase where M.Key == String,
```

Comment:
What is the point of this being generic with all of the constraints baked in like this?

Reply:
So I could test against a normal `Map<String, SwiftStringObject?>` collection and a `AnyMap()` collection